webpack/assets/javascripts/react_app/components/Bookmarks/Bookmarks.js

```diff
@@ -24,21 +24,26 @@ class Bookmarks extends React.Component {
   }
 
   handleNewBookmarkClick() {
-    const { showModal, modalClosed, modalOpened, searchQuery } = this.props;
+    const {
+      isModalOpen,
+      setModalClosed,
+      setModalOpen,
+      setQuery,
+      searchQuery,
+    } = this.props;
 
-    if (showModal) {
-      modalClosed();
+    if (isModalOpen) {
+      setModalClosed();
     } else {
-      modalOpened(searchQuery);
+      setQuery(searchQuery);
+      setModalOpen();
     }
   }
```

Comment:
Something really weird is going on,
looks like there was a problem with the old modal here, so the logic to close that modal was added,
I tested it with the new modal and in my opinion this method should be removed and instead on line 67,
it should look like: 
```js
<MenuItem
    key="newBookmark"
    id="newBookmark"
    onClick={this.props.setModalOpen}
>
```

the second issue that I found is why the `searchQuery` that comes from the `autocomplete` searchBar's state 
should be saved again in the bookmarks state? we already got it as a prop, so `setQuery` is redundant too.
IMO we could pass the `searchQuery` to the `BookmarkForm` or use a selector there to select it from the autocomplete.

Reply:
whew, that was a journey.  Updating, thanks @laviro :)

Reply:
The selector in `webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteSelectors.js` that gives us the search query requires 3 args: `state`, `id`, and `ownProps`.  `id` is always `searchBar`.   `state` must contain an `autoComplete` key with subkey `id` and the same value `searchBar`;  and `ownProps` is passed through all the selectors in that file.  It broke all the integration tests at first because I didn't have the `searchBar` value in both places that it needed to be.  Anyway, I don't really want to spend much more time refactoring this but I at least got it to a place where there's a single source of truth for the search query.

Reply:
found a solution, see below